packages/xo-server/src/api/test.js

```diff
@@ -86,3 +89,32 @@ copyVm.resolve = {
   vm: ['vm', 'VM'],
   sr: ['sr', 'SR'],
 }
+
+// -------------------------------------------------------------------
+
+export async function xapiPoolChange({
+  currentPoolObject,
+  newPoolHostname,
+  newPoolObject,
+}) {
+  const xapi = this.getXapi(currentPoolObject)
+  const { _pool: currentPool } = xapi
+
+  xapi._setUrl({ ...xapi._url, hostname: newPoolHostname })
+  const [newPool] = await Promise.all([
+    // used to trigger the "update" event
+    xapi.getAllRecords('pool').then(array => array[0]),
```

Comment:
This won't trigger and `updater`…

Reply:
Should i trigger manually a `session.open`?

Reply:
Nope.

Why do you need to trigger this?

Reply:
`_setUrl` don't open a new session. So, the XAPI pool will not be changed.

Reply:
The goal of this test is to make sure that the change is automatically detected and properly handled, we should not have to do anything special.

Reply:
Ok then `_setUrl` should be an `async`? Because it should do some stuff like updating the xapi pool ? Or you have a better idea?

Reply:
Nope, this is a test, we will not change the internal architecture of `xen-api` just for this.

You can wait for a the `next` finish event which will indicate that new events has been received, the state should have updated to pool change by then.

Reply:
Can you please give me an example of the listener? Because i don't see how to do it.